packages/language-server/src/plugins/typescript/SnapshotManager.ts

```diff
@@ -103,12 +110,19 @@ export class SnapshotManager {
 
     constructor(
         private globalSnapshotsManager: GlobalSnapshotsManager,
-        private projectFiles: string[],
         private fileSpec: TsFilesSpec,
-        private workspaceRoot: string
+        private workspaceRoot: string,
+        projectFiles: string[],
+        useCaseSensitiveFileNames = ts.sys.useCaseSensitiveFileNames
     ) {
         this.onSnapshotChange = this.onSnapshotChange.bind(this);
         this.globalSnapshotsManager.onChange(this.onSnapshotChange);
+        this.documents = new FileMap(useCaseSensitiveFileNames);
+        this.projectFileToOriginalCasing = new FileMap(useCaseSensitiveFileNames);
```

Comment:
Not quite understand the use case if this map here - it's not set with `original,case_insensitive` or similar, it's set with `originalFileName, originalFileName` in all places where it's called

Reply:
It's essentially `Map<case_insensitive, original>`. I think I can change it to directly use `Map<case_insensitive, original>` so it's more clear on its use case.

Reply:
What I meant is that everywhere where `set` is invoked you call it with the same variable for both parameters

Reply:
Yes. Because the `FileMap` will transform the casing for the key but not the value. That is indeed confusing as it keeps two different things.